client/tests/webdriver/specs/customFields.spec.js

```diff
@@ -0,0 +1,274 @@
+      )
+    })
+
+    it("Should toggle the correct invisible fields", () => {
+      const button = CreateReport.getEngagementTypesButtonByName(
+        TESTED_ENGAGEMENT_BUTTON
+      )
+      button.waitForExist()
+      button.waitForDisplayed()
+      button.click()
+      CreateReport.fieldsToggledVisibilityByTrainButton.forEach(
+        nowVisisbleField => {
+          nowVisisbleField.waitForExist()
+          nowVisisbleField.waitForDisplayed()
+        }
+      )
+
+      CreateReport.fieldsNotToggledVisibilityByTrainButton.forEach(
+        stillInvisField => {
+          expect(stillInvisField.isExisting()).to.equal(false)
+        }
+      )
+    })
+
+    it("Should persist previous valid data when toggling field's visibility", () => {
+      CreateReport.numberTrainedField.setValue(validNumberInput)
+      const button = CreateReport.getEngagementTypesButtonByName(
+        TESTED_ENGAGEMENT_BUTTON
+      )
+      // make invisible
+      button.click()
+      CreateReport.numberTrainedFormGroup.waitForExist({ reverse: true })
+      // make visible
+      button.click()
+      CreateReport.numberTrainedFormGroup.waitForExist()
+
+      expect(CreateReport.numberTrainedField.getValue()).to.equal(
+        validNumberInput
+      )
+    })
+
+    it("Should not persist previous invalid data when toggling field's visibility", () => {
+      CreateReport.numberTrainedField.setValue(inValidNumberInput)
+      CreateReport.numberTrainedHelpText.waitForExist()
+      // Actually see the validation warning
+      expect(CreateReport.numberTrainedHelpText.getText()).to.equal(
+        "Number trained must be greater than or equal to 1"
+      )
+      const button = CreateReport.getEngagementTypesButtonByName(
+        TESTED_ENGAGEMENT_BUTTON
+      )
+      // make invisible
+      button.click()
+      CreateReport.numberTrainedFormGroup.waitForExist({ reverse: true })
+      // make visible
+      button.click()
+      CreateReport.numberTrainedFormGroup.waitForExist()
+
+      expect(CreateReport.numberTrainedField.getValue()).not.be.equal(
+        inValidNumberInput
+      )
+      expect(CreateReport.numberTrainedField.getValue()).to.be.equal("")
+    })
+
+    it("Should validate visible field", () => {
+      CreateReport.numberTrainedField.setValue(inValidNumberInput)
+      CreateReport.numberTrainedHelpText.waitForExist()
+      CreateReport.submitForm()
+      CreateReport.waitForAlertToLoad()
+
+      expect(CreateReport.alert.getText()).to.include(
+        "Number trained must be greater than or equal to 1"
+      )
+    })
+
+    it("Should show valid visible field", () => {
+      // we are already at show page
+      expect(CreateReport.numberTrainedFieldShowed.getText()).to.include(
+        validNumberInput.toString()
+      )
+    })
+
+    it("Should not validate invisible field", () => {
+      CreateReport.editButton.click()
+      CreateReport.form.waitForExist()
+      CreateReport.form.waitForDisplayed()
+
```

Comment:
Maybe add a comment: number trained is invalid, to make the tests easier to read.
I find it hard to see what the state is in each test. Which value a field has, or whether the train button is on or off ....